src/Microsoft.Win32.SystemEvents/tests/SystemEvents.CreateTimer.cs

```diff
@@ -104,33 +101,30 @@ public void ConcurrentTimers()
             {
                 if (PlatformDetection.IsFullFramework)
                 {
-                    // desktop has a bug where it will allow EnsureSystemEvents to proceed without actually creating the HWND
+                    // netfx has a bug where it will allow EnsureSystemEvents to proceed without actually creating the HWND
                     SystemEventsTest.WaitForSystemEventsWindow();
                 }
 
+                // Create all the timers
+                var timers = new List<IntPtr>();
                 for (int i = 0; i < NumConcurrentTimers; i++)
                 {
-                    lock (timersSignalled)
-                    {
-                        timersSignalled[SystemEvents.CreateTimer(TimerInterval)] = false;
-                    }
+                    timers.Add(SystemEvents.CreateTimer(TimerInterval));
                 }
 
-                Assert.True(elapsed.WaitOne(TimerInterval * SystemEventsTest.ExpectedEventMultiplier));
+                // Wait for them all to fire
+                Assert.True(elapsed.Wait(TimerInterval * SystemEventsTest.ExpectedEventMultiplier));
 
-                lock (timersSignalled)
+                // Delete them all
+                foreach (IntPtr timer in timers)
                 {
-                    foreach (var timer in timersSignalled.Keys.ToArray())
-                    {
-                        Assert.True(timersSignalled[timer]);
-                        SystemEvents.KillTimer(timer);
-                    }
+                    Assert.True(timersSignaled.TryGetValue(timer, out _));
+                    SystemEvents.KillTimer(timer);
                 }
             }
             finally
             {
                 SystemEvents.TimerElapsed -= handler;
-                elapsed.Dispose();
```

Comment:
Why not dispose anymore?

Reply:
It's not strictly necessary (i.e. there aren't actually any resources to clean up in this case), and I was concerned about some subsequent handler coming along and trying to use it after it's been disposed, but after the dust here settled it's actually not an issue anymore, so I could put it back.

Reply:
I don't mind either way, it just catches the eye whenever there's something I know is disposable that the code makes no plan to dispose. Plus, the implementation of the disposable thing could in theory change, making disposing more important.